driver/src/main/scala/za/co/absa/hyperdrive/driver/IngestionDriver.scala

```diff
@@ -61,11 +66,19 @@ private[driver] class IngestionDriver {
 
   private def getStreamWriter(conf: Configuration): StreamWriter = StreamWriterAbstractFactory.build(conf)
 
+  private def getIngestionFinalizer(conf: Configuration): IngestionFinalizer = IngestionFinalizerAbstractFactory.build(conf)
+
   private def printConfiguration(configuration: Configuration): Unit = {
     import scala.collection.JavaConverters._
     configuration
       .getKeys
       .asScala
       .foreach(key => logger.info(s"\t$key = ${configuration.getProperty(key)}"))
   }
+
+  private def applyDefaults(configuration: Configuration): Unit = {
+    DefaultConfiguration.values.foreach {
+      case (key, value) => if (!configuration.containsKey(key)) configuration.addProperty(key, value)
```

Comment:
in a `match case` situation you would put the if in from of the `=>`. Is this different?
Because my brain finds it non-standard, took me a while to absorb

Reply:
Ah yes totally forgot about that. Actually if I look at it again, it's probably more concise to write
```
DefaultConfiguration.values
      .filter{case (key, _) => !configuration.containsKey(key)}
      .foreach{case (key, value) => configuration.addProperty(key, value)}
```